### PR TITLE
[MPDX-9655] - Allow one time Savings Fund Transfers to be future dated

### PR DESCRIPTION
**Pull request by wjames111**
## Description

- Replaces the disabled (grayed-out) Transfer Date field on one-time Savings Fund Transfers with an editable date picker, so users can schedule a one-time transfer for today or a future date
- Sends the selected date to the API via the new required `transactedAt` argument on the `CreateTransfer` mutation
- Adds a one-time-specific validation message ("Transfer date cannot be in the past") — previously a past date showed the recurring-transfer error message
- Adds tests covering the enabled date field, future-date submission (including `transactedAt` in the mutation variables), and past-date validation

**Dependent on:** CruGlobal/mpdx_api#3342 (adds the required `transactedAt` argument to `TransferCreateMutation`). GraphQL codegen in CI will fail until that PR is deployed to the staging API. Note also that since `transacted_at` is required on the API side, deploying the API change before this PR would break one-time transfers on the current UI — release order needs coordination.

Jira: [MPDX-9655](https://jira.cru.org/browse/MPDX-9655)

## Testing

- Go to HR Tools → Staff Saving Fund → Transfers
- Click a balance card to open the New Fund Transfer modal
- With the "One Time" schedule selected, check that the Transfer Date field is enabled (no longer grayed out) and defaults to today
- Pick a future date, fill in To Account, Amount, and Note, and submit — check that the transfer is created with the selected date
- Pick a past date and check that the "Transfer date cannot be in the past" validation error appears and Submit is disabled
- Switch to "Monthly" and check that the recurring transfer date/end date behavior is unchanged

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels (_Add the label "Preview" to automatically create a preview environment_)
- [x] I have run the Claude Code `/pr-review` command locally and fixed any relevant suggestions
- [ ] I have requested a review from another person on the project
- [ ] I have tested my changes in preview or in staging
- [x] I have cleaned up my commit history

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.test.tsx [hunk 1]
```diff
@@ -431,6 +431,44 @@ describe('TransferModal', () => {
       ).toBeInTheDocument();
     });
 
+    it('should allow editing the transfer date for one-time transfers', async () => {
```
Comment (wjames111) [inline src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.test.tsx:434]: [Medium] **Date-edge cases untested on the newly-editable field** (Testing + Data Integrity, supported by Architecture & UX after debate)

The field went from `disabled` to an editable, clearable, required input, but no test covers:

1. **Cleared/empty date** — every `userEvent.clear(transferDate)` in the suite is immediately followed by a `type(...)`. Nothing asserts that clearing the date surfaces "Transfer date is required" and disables submit. The runtime behavior was verified correct during review, so this is a regression-exposure gap, not a bug — but a dropped `.required()` would currently be caught by nothing.
2. **Invalid/garbage date** — takes a *different* yup branch: an invalid Luxon DateTime is truthy, slips past `required()`, and is blocked only by the `start-date` comparison (`selected >= minimum` is false for invalid dates).

Suggested addition:

```tsx
it('should require a transfer date for one-time transfers when cleared', async () => {
  const { getByRole, getByLabelText, findByText } = render(<Components />);
  const transferDate = getByLabelText(/transfer date/i);
  userEvent.clear(transferDate);
  userEvent.tab();
  expect(await findByText('Transfer date is required')).toBeInTheDocument();
  await waitFor(() =>
    expect(getByRole('button', { name: /submit/i })).toBeDisabled(),
  );
});
```

plus a variant typing an unparseable date and asserting submit stays disabled. (Minor completeness add, conceded to 3.5 in debate: a user-*typed* today's date is also unasserted, though the equality boundary is already exercised by the default-submit test and typed parsing by the future-date test.)

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx [hunk 2]
```diff
@@ -219,6 +222,7 @@ export const TransferModal: React.FC<TransferModalProps> = ({
             sourceFundTypeName: transferFrom,
             destinationFundTypeName: transferTo,
             description: note.trim(),
+            transactedAt: convertedTransferDate,
```
Comment (wjames111) [inline src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx:222]: [Suggestion] **Type-honesty + guarded serialization** (Data Integrity, severity revised down 6.5 → ~4.0 in debate after the crash was proven unreachable)

`transferDate` is typed non-nullable (`DateTime<boolean>`, line 47), but the now-editable picker can emit `null` (clear / empty blur) or an invalid DateTime, and `convertedTransferDate = transferDate.toISO() ?? ''` (line 193) now feeds this required `ISO8601DateTime!` argument. The crash/empty-string paths are unreachable today — Formik's `submitForm` re-validates and never calls `onSubmit` with a validation error (note: `!isValid` button gating alone would *not* protect the first render, since `isValid` starts `true` without `validateOnMount`).

Hardening suggestion so correctness doesn't depend on validation wiring:

```tsx
// interface
transferDate: DateTime<boolean> | null;

// handleSubmit
const convertedTransferDate =
  transferDate && transferDate.isValid ? transferDate.toISO() : '';
```

(or throw instead of `''` so a future gating regression fails loudly rather than sending an empty string to the API). Verified in debate that widening the type does not ripple into the recurring branch — the `end>start` test already null-guards and `initialValues` defaults to `getToday()`.

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx [hunk 3]
```diff
@@ -104,9 +104,12 @@ const transferSchema = (locale: string) =>
         }
 
         return this.createError({
-          message: i18n.t(
-            'Recurring transfers must start at least one day in the future',
-          ),
+          message:
+            schedule === ScheduleEnum.OneTime
```
Comment (wjames111) [inline src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx]: [Suggestion] **Three message-producing branches in one yup test** (Architecture)

The `start-date` test now carries `isEditing` → "earlier than {date}", one-time → "cannot be in the past", and recurring → "one day in the future", interleaved with the schedule-dependent `minimum` computation. Logic is sound and test-verified; optional extraction of `minimumTransferDate(schedule)` / `pastDateMessage(schedule)` helpers would keep it readable as more schedule types (the enum already has `Annually`) arrive. Not required for merge.

One behavioral note from debate worth confirming intentionally: `isEditing` short-circuits *before* the one-time/recurring branch, so an edited transfer never sees the new "cannot be in the past" message — moot today since only recurring transfers are editable, but worth remembering if one-time editing is ever added.

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx [hunk 4]
```diff
@@ -455,11 +459,21 @@ export const TransferModal: React.FC<TransferModalProps> = ({
                   <Grid container spacing={2}>
                     {schedule === ScheduleEnum.OneTime ? (
                       <Grid size={12}>
-                        <TextField
-                          fullWidth
+                        <CustomDateField
```
Comment (wjames111) [inline src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx:478]: [Suggestion] **Two low-cost follow-ups on the new date field** (Architecture + UX)

1. This `CustomDateField` block is now byte-identical to the recurring branch's transferDate field (only the Grid `size` differs). Optionally hoist a single field above the conditional to avoid maintaining the wiring twice.
2. The calendar doesn't constrain past dates — `disablePast`/`minDate` would gray out invalid days instead of erroring after the fact. `CustomDateField`'s desktop path can't honor those props today (they'd be silently dropped), so this is follow-up-ticket material for the shared component (would benefit the recurring branch too), not this PR.

UX notes from review (no action needed): desktop surfaces the required-error only on blur (brief "disabled submit, no visible reason" window — consistent with the existing form model); mobile's explicit Clear action surfaces it immediately.

**Review by wjames111 — COMMENTED** [review]
## Multi-Agent Code Review (standard mode)

**Verdict: APPROVED WITH SUGGESTIONS** — 0 blockers, 1 medium-priority finding, 7 informational suggestions.

**Risk:** 5/10 (MEDIUM) — mutation contract change with a cross-repo deploy dependency (CruGlobal/mpdx_api#3342, already documented in the PR description).

**Dependency impact:** LOW — `TransferModal` has one production render site; `CreateTransfer` has exactly one call site, updated in this PR. No breaking changes.

**Agents:** Architecture, Testing, Data Integrity, UX, Standards, Financial Reporting (domain) + gap review + dependency analysis, with cross-examination debate and rebuttal rounds. Standards checklist: full PASS. Debate rounds *downgraded* two findings with evidence: the initially-flagged "latent null-date crash" was proven unreachable (Formik's `submitForm` validates before `onSubmit`), and the today-boundary was shown to already be exercised end-to-end by the default-submit test.

**Timezone adjudication (asked by 4 agents, resolved with evidence):** `transferDate.toISO()` serializes local start-of-day *with* the local offset, and the backend takes `.to_date` in that offset — the user's chosen calendar date survives in every timezone. The default-today path is not a regression vs the old server-side default.

---

### Findings on Related Files (Not in This PR)

**[Medium] Downstream display of future-dated one-time transfers** — flagged by the Financial Reporting agent, sharpened in debate
- This PR is the first code path that lets a one-time transfer carry a future `transactedAt`, which *activates* latent assumptions in pre-existing files: `src/components/HrTools/SavingsFundTransfer/Helper/getStatus.ts:5-6` returns **Complete** for any non-recurring row regardless of date; `Helper/filterTransfers.ts` routes any row with a transaction to History; `TransfersPage/TransfersPage.tsx:186-199` hardcodes Upcoming rows as Monthly (a one-time row landing there would render amount 0 / undefined date).
- Which of these manifests depends on whether the backend materializes the transaction immediately or at the transacted date.
- Recommended action: confirm backend behavior with the mpdx_api team and file a follow-up ticket for the display pipeline. Informational — does not block this PR.

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.test.tsx (hunk 1 above)
Comment (wjames111) [reply src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.test.tsx:434]: 🤖 Fixed in 8900f16: added tests for cleared/empty date (required error + disabled submit) and invalid date (start-date branch, submit blocked, mutation not fired).

_Posted by Claude Code via /agent-fix._

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx (hunk 2 above)
Comment (wjames111) [reply src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx:222]: 🤖 Fixed in f6d7602: widened transferDate to DateTime<boolean> | null and guarded the ISO serialization with an isValid check.

_Posted by Claude Code via /agent-fix._

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx (hunk 3 above)
Comment (wjames111) [reply src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx]: 🤖 Fixed in 62aa3ad: extracted minimumTransferDate(schedule) and pastDateMessage(schedule) module-scope helpers (static i18n keys preserved); validation behavior unchanged.

_Posted by Claude Code via /agent-fix._

File: src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx (hunk 4 above)
Comment (wjames111) [reply src/components/HrTools/SavingsFundTransfer/TransferModal/TransferModal.tsx:478]: 🤖 Fixed in 3135265: hoisted the duplicated Transfer Date CustomDateField into a single block (Grid size 12/6 by schedule); disablePast/minDate support in the shared component remains a follow-up ticket.

_Posted by Claude Code via /agent-fix._

**Comment by zweatshirt** [comment]
Seems the transfer (if future dated) works, but fails to display in the upcoming transfers table

**Comment by wjames111** [comment]
@zweatshirt yeah workin on that now, I'll let you know once fixed.
